components/common/typescript.gni

```diff
@@ -119,9 +124,13 @@ template("transpile_web_ui") {
       gen_output_dir = "$root_gen_dir/brave/web-ui-$resource_name"
     }
 
-    # sole output is a GRD that includes all the resources
+    # Output is a GRD that includes all the resources, and a bundle for each
+    # entry point.
     grd_name = "$resource_name.grd"
     outputs = [ "$gen_output_dir/$grd_name" ]
+    foreach(entry_point, invoker.entry_points) {
+      outputs += [ "$gen_output_dir/" + entry_point[0] + ".bundle.js" ]
```

Comment:
This seems tricky because we're guessing what the webpack outputs will be, and in many cases it's more files than just the x.bundle.js. It would not include extra .js chunks, or image assets, for example. What problem were you running in to that this is neccessary? In my experience, the .grd file was enough since it includes all the files output from webpack, and the grit target which has the .grd as an input also gets all those webpack outputs as inputs automatically. If any of those change, then the grit target would re-build. And those can only change because of webpack re-running, which happens because of any input TS files changing, etc...

Reply:
Our current webpack doesn't seem to support any chunking (at least, in my limited attempts to get it to chunk stuff). If you add multiple entry points, common stuff is duplicated in each bundle.\

You're right it won't catch any image assets though (or chunks if we support them in the future). 

The problem I was trying to solve is that I want to copy the webpack output to a different folder ($brave_resources_extra_grdps_path/leo/) and in order to do that the bundle.js has to be listed as output from another build target.

Even if I was able to output the bundle directly to the correct folder, I'd need to know the output assets so I can add them to the `static_grdp` [here](https://github.com/brave/brave-core/pull/18156/files#diff-d0796ca4beef766fd701f7d4799d6d3336924c9fc8fd00226b92936b4703a6e4R101).

Reply:
I've updated the `transpile_web_ui` script to optionally output a `grdp` file which we can include in `chrome://resources` in [this commit](https://github.com/brave/brave-core/pull/18156/commits/ef407a047b5aeab2fc71a35c4a2c676931711b3c). What do you think of this approach?